.github/workflows/cleandocker.yml

```diff
@@ -0,0 +1,29 @@
+name: CleanDocker
+on:
+  pull_request:
+    types: [closed]
+
+jobs:
+  clean-docker:
+    name: Clean Docker images for PR
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_USER: ${{ secrets.DOCKER_USER }}
+      DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+      GITHUB_CONTEXT: ${{ toJson(github) }}
+    steps:
+    - name: Get Commits and delete
+      run: |
+        COMMITS_URL=$(echo $GITHUB_CONTEXT | jq -r '.event.pull_request._links.commits.href')
+        COMMITS=$(curl $COMMITS_URL | jq -r '.[] | .sha')
+        echo "***Commits***"
+        echo $COMMITS
+        echo "******"
+
+        TOKEN=$(curl -s -H "Content-Type: application/json" -X POST --data '{"username": "'"$DOCKER_USER"'", "password": "'"$DOCKER_PASSWORD"'"}' https://hub.docker.com/v2/users/login/ | jq -r .token)
+        for commit in $COMMITS
+        do
+          echo "Deleting docker image ${commit:0:7}"
+          curl -X DELETE -s -H "Authorization: JWT ${TOKEN}" https://hub.docker.com/v2/repositories/mattermost/bifrost/tags/"${commit:0:7}"/ || true
+        done
+        echo done
```

Comment:
Would suggest dropping this, in favor of a Lifecycle Policy in the Docker Registry part.